jbpm-task-assigning/jbpm-task-assigning-core/src/test/java/org/jbpm/task/assigning/BaseTaskAssigningTest.java

```diff
@@ -0,0 +1,155 @@
+        SolverConfig config = solverFactory.getSolverConfig();
+        config.setSolutionClass(TaskAssigningSolution.class);
+        config.setEntityClassList(Arrays.asList(TaskOrUser.class, Task.class));
+        config.setScoreDirectorFactoryConfig(new ScoreDirectorFactoryConfig().withScoreDrls("org/jbpm/task/assigning/solver/taskAssigningScoreRules.drl"));
+        return solverFactory;
+    }
+
+    protected Solver<TaskAssigningSolution> createDaemonSolver() {
+        SolverFactory<TaskAssigningSolution> solverFactory = createSolverFactory();
+        solverFactory.getSolverConfig().setDaemon(true);
+        return solverFactory.buildSolver();
+    }
+
+    protected Solver<TaskAssigningSolution> createNonDaemonSolver(long millisecondsSpentLimit) {
+        SolverFactory<TaskAssigningSolution> solverFactory = createSolverFactory();
+        solverFactory.getSolverConfig().setTerminationConfig(new TerminationConfig().withMillisecondsSpentLimit(millisecondsSpentLimit));
+        return solverFactory.buildSolver();
+    }
+
+    protected TaskAssigningSolution readTaskAssigningSolution(String resource) throws IOException {
+        int index = resource.lastIndexOf("/");
+        String prefix = resource;
+        if (index >= 0) {
+            prefix = resource.substring(index + 1);
+        }
+        File f = File.createTempFile(prefix, null);
+        InputStream resourceAsStream = getClass().getResourceAsStream(resource);
+        FileUtils.copyInputStreamToFile(resourceAsStream, f);
+        XStreamSolutionFileIO<TaskAssigningSolution> solutionFileIO = new XStreamSolutionFileIO<>(TaskAssigningSolution.class);
+        return solutionFileIO.read(f);
+    }
+
+    private static void appendln(StringBuilder builder) {
+        builder.append('\n');
+    }
+
+    private static void appendln(StringBuilder builder, String text) {
+        builder.append(text);
+        appendln(builder);
+    }
+
+    public static void printSolution(TaskAssigningSolution solution, StringBuilder builder) {
+        solution.getUserList().forEach(taskOrUser -> {
+            appendln(builder, "------------------------------------------");
+            appendln(builder, printUser(taskOrUser));
+            appendln(builder, "------------------------------------------");
+            appendln(builder);
+            Task task = taskOrUser.getNextTask();
+            while (task != null) {
+                builder.append(" -> ");
+                appendln(builder, printTask(task));
+                task = task.getNextTask();
+                if (task != null) {
+                    appendln(builder);
+                }
+            }
+            appendln(builder);
+        });
+    }
+
+    public static String printSolution(TaskAssigningSolution solution) {
+        StringBuilder builder = new StringBuilder();
+        printSolution(solution, builder);
+        return builder.toString();
+    }
+
+    public static String printUser(User user) {
```

Comment:
Suggestion: these printing methods could be actually implemented as `toString()` of the particular classes they print information about.

Reply:
I prefer to not by now. I really came up with this "printXXX" methods for being able to see the test results in a tabular manner, etc. And I can change them free if I have particular needs related with the test.